### DISCUSSION TITLE
Automatic component import

### DISCUSSION THREAD

**Discussion by HenrijsS** [sveltejs/kit#3032] [Ideas]
I think a good idea to break out even more of the import hell that React has is to utilize automatic component import.

Nuxt.js already does this perfectly: [as seen here](https://nuxtjs.org/docs/features/component-discovery)

You could also use the same syntax like Nuxt, for example:
A component in `src/lib/header/Navigation.Svelte` could resolve to `<HeaderNavigation />`. 
Thus removing the need to import it manually if a component/page includes several other components.

Is this something considered?

**Comment by dominikg** [comment]
Personally i'm not a fan of these magic imports. They prevent static drpendency tracking and especially camelCase to path separators reads like a nightmare to support/unwind

It may work with https://github.com/antfu/unplugin-auto-import buuut you'd need to see how to use it from a svelte preprocessor as the imports need to be present when svelte.compile is called.

Maybe @userquin has an idea how it could be implemented.

But tbh even if it works i'd not recommend using it. IDE support should autogenerate imports for you so this very minor dx improvement isn't worth the loss of information in file

**Reply by hamedbaatour** [reply]
valid points but honestly coming from Nuxt 3 this seems like something I can't live without anymore, having worked in the past with large apps with many components I realized how quickly that the import hell becomes real!

the mental load of scrolling through the first dozens of lines of code and seeing those imports on a daily basis is just tiring
remembering to actually hit import every single time and relying on IDEs support for importing and sorting those imports (using different plugins / sometimes configuring linters ) is also just unnecessary work that could have been avoided from the beginning by removing imports altogether. 

I know that static dependency tracking is essential but can we have this at least optional in `svelte.config.js` or as a separate package so more developers coming from Nuxt adopt Sveltekit quickly and make their transition more seamless as auto imports truly add up to the developer experience of the framework.

**Reply by ZetiMente** [reply]
> valid points but honestly coming form Nuxt 3 this seems like something I can't live without anymore, having worked in the past with large apps with many components I realized how quickly that the import hell becomes real!
> 
> the mental load of scrolling through the first dozens of lines of code and seeing those imports on a daily basis is just tiring remembering to actually hit import every single time and relying on IDEs support for importing and sorting those imports (using different plugins / sometimes configuring linters ) is also just unnecessary work that could have been avoided from the begging by removing imports all together.
> 
> I know that static dependency tracking is important but can we have this at least optional in `svelte.config.js` or as a separate package so more developers coming from Nuxt adopt Sveltekit quickly and make their transition more seamless as auto imports truly add up to the developer experience of the framework.

@hamedbaatour I asked Rich for it but he said no and directed me here: https://github.com/yuanchuan/sveltekit-autoimport

I'm still a fan of it but he seems to think it makes it harder to find stuff.
